sky/sky_logging.py

```diff
@@ -25,18 +27,60 @@ def format(self, record):
         return msg
 
 
+_root_logger = logging.getLogger('sky')
+_default_handler = None
+echo = print
```

Comment:
Some Q's for discussion.

Why do we not want the callers to directly use `print()`? For devs who aren't aware of this module's `echo`, it's easy to write `print()` and then `silent()` becomes a leaky abstraction. 

If `_get_spot_jobs` is the only caller of `silent()`, can we do away with `silent()` and use the previous suppress_output()...?

Reply:
> The reason we have to abandon the original `suppress_output` is that, it will redirect all the stdout and stderr to the devnull, making the `rich.console.status` stop working.

Are you saying `suppress_output()` makes `core.spot_queue()`'s usage of `rich.console.status` stop working? See the other comment about prog. APIs not using spinners.